integration/common/openlineage/common/provider/dbt/local.py

```diff
@@ -101,10 +104,45 @@ def __init__(
 
         self.target = target
         self.project_name = dbt_project["name"]
+        self.models = models or []
+        self.selector = selector
         self.profile_name = profile_name or dbt_project.get("profile")
         if not self.profile_name:
             raise KeyError(f"profile not found in {dbt_project}")
 
+    @property
+    def _use_extended_job_name(self):
+        # The name of the job published on the OpenLineage events for the dbt-ol run wrapper
+        # should include extended attributes (profile and model) only if this environment
+        # variable is set, in order to prevent breaking back-compatibility.
+        var = os.getenv("OPENLINEAGE_DBT_USE_EXTENDED_JOB_NAME", "false").lower()
+        if var in ("true", "1"):
+            return True
+        if var in ("false", "0"):
+            return False
+
+        raise ValueError(f"Invalid value for OPENLINEAGE_DBT_USE_EXTENDED_JOB_NAME: {var}")
+
+    @staticmethod
+    def _sanitize_job_name_component(s: str) -> str:
+        return re.sub(r"[^a-zA-Z0-9_]", "_", s)
+
+    def job_name(self) -> str:
+        job_name = f"dbt-run-{self.project_name}"
+        if not self._use_extended_job_name:
+            return job_name
+
+        if self.profile_name:
+            job_name = f"{job_name}-{self.profile_name}"
+        if self.models:
+            job_name = f"{job_name}-" + "-".join(
+                [self._sanitize_job_name_component(model) for model in self.models]
+            )
+        elif self.selector:
```

Comment:
Do you think that if chain covers all the potential cases when we want to have extended job name? 

I'm not that up to date with `dbt` lately so I don't know - we don't have to cover everything here, but if there's something else I'd leave a comment there.

Reply:
Talking from the experience of how we use dbt in Booking, I think that a combination of profile + model list/selector should be sufficient for us to uniquely identify a task. But I'm not sure if this may be the case for everyone. I've just pushed a comment for this property that explains the `job_name` construction strategy and invites anyone with more clue than me to contribute if they feel like more identifiers are needed.